packages/common/src/users/HubUser.ts

```diff
@@ -0,0 +1,125 @@
+import { IEntityEditorContext, IHubUser } from "../core/types";
+import { IArcGISContext } from "../ArcGISContext";
+import { getEditorConfig } from "../core/schemas/getEditorConfig";
+import { IEditorConfig } from "../core/schemas/types";
+import { IWithEditorBehavior } from "../core/behaviors/IWithEditorBehavior";
+import { enrichEntity } from "../core/enrichEntity";
+import { cloneObject } from "../util";
+import { UserEditorType } from "./_internal/UserSchema";
+
+export class HubUser implements IWithEditorBehavior {
+  protected context: IArcGISContext;
+  protected entity: IHubUser;
+  protected isDestroyed = false;
+
+  private constructor(user: IHubUser, context: IArcGISContext) {
+    this.entity = user;
+    this.context = context;
+  }
+
+  /**
+   * Create an instance from a IHubUser object
+   * @param json - JSON object to create a HubProject from
+   * @param context - ArcGIS context
+   * @returns
+   */
+  static fromJson(json: Partial<IHubUser>, context: IArcGISContext): HubUser {
+    // merge what we have with the default values
+    const pojo = this.applyDefaults(json);
+    return new HubUser(pojo, context);
+  }
+
+  private static applyDefaults(partialUser: Partial<IHubUser>): IHubUser {
+    // TODO: move to defaults
+    const DEFAULT_USER = {};
+    return { ...DEFAULT_USER, ...partialUser } as IHubUser;
+  }
+
+  /**
+   * Method that returns the entity as a JSON object
+   * We have this on the EntityItem class, but we don't implement that here
+   * @returns
+   */
+  toJson(): IHubUser {
+    return cloneObject(this.entity);
+  }
+
+  /**
+   * Save the HubUser to the backing store
+   * TODO: fill this out
+   */
+  async save(): Promise<void> {
+    if (this.isDestroyed) {
+      throw new Error("HubUser is already destroyed.");
+    }
+
+    // 1. update user hub settings
+    await this.context.updateUserHubSettings(this.entity.settings);
```

Comment:
I think this is perfectly fine as-is, but now it just has me wondering about how we're doing things for other entities...basically I'm wondering if this should happen in `save` or in `fromEditor` 🤔 

In projects, for example, we make the request to upsert/remove the configured featured image directly in `fromEditor` before calling `save`.

In general (and as we've discussed), I think there's a lot of ambiguity in this whole `EntityEditor` system, so I'm not actually sure what to suggest or if one way is better than the other.